sage-ios/SAGE/SAGE/ProfileViewController.swift

```diff
@@ -139,6 +164,36 @@ class ProfileViewController: UITableViewController {
         }
     }
     
+    func setButtonVisibility() {
```

Comment:
this logic should definitely go inside the view itself, you can probably even put it in the setupWithUser method and pass in some argument like perspectiveUser: in order to deal with what buttons show or dont show. then you dont have to call this manually every time setup is called. the Edit Profile visitibility logic can also go in the setupWithUser method this way too